earthaccess/results.py

```diff
@@ -107,6 +109,38 @@ def get_umm(self, umm_field: str) -> Union[str, Dict[str, Any]]:
         """
         return self["umm"].get(umm_field, "")
 
+    def get_doi(self) -> Optional[str]:
+        """Retrieve the Digital Object Identifier (DOI) for this collection.
+
+        Returns:
+            A collection's DOI information or None.
+        """
+        doi = self["umm"].get("DOI", {})
+        if isinstance(doi, dict):
+            return doi.get("DOI", None)
+        return None
+
+    def get_citation(self, format: str, language: str) -> Optional[str]:
```

Comment:
Let's drop the `get_` prefix here too, use `|` instead of `Optional`, and use kw-only params since they are both type `str`, which makes it too easy to pass them in the wrong order without forcing keyword usage:

```suggestion
    def citation(self, *, format: str, language: str) -> str | None:
```